Assets/Scripts/MusicManager.cs

```diff
@@ -162,6 +159,16 @@ private void CheckQuestsCompleted()
             QuestsCompleted = dialogueQuests;
     }
 
+    /// <summary>
+    /// Sets variables to start fading the music
+    /// </summary>
+    private void BeginMusicFade()
+    {
+        isVolumeChanging = true;
+        isFadingOut = true;
```

Comment:
what's the difference between isVolumeChanging and IsFadingOut? Explain it to me in your reply please.

Reply:
I thought the name change would make that clearer. IsVolumeChanging just keeps track of whether the music is fading at all. IsFadingOut specifically notes whether it's fading out (which it is, at first) or fading back in. This allows me to only run the fade function when the music is preparing to change, and also know when it should fade back in to normal volume.